File: done/credit/credit.c
#include <cs50.h>
#include <stdio.h>

int check_sum(long n);
int check_length(long n);
long f_two(long n);

int main(void)
{
    // get card num and first two digits
    long n = get_long("Number: ");
    int t = f_two(n);

    // check card sum & length
    int l = check_length(n);
    if (check_sum(n) != 1 || (l != 13 && l != 15 && l != 16))
    {
        printf("INVALID\n");
    }

    // print card type
    else if (t == 34 || t == 37)
    {
        printf("AMEX\n");
    }

    else if (t > 50 && t < 56)
    {
        printf("MASTERCARD\n");
    }

    else if (t / 10 == 4)
    {
        printf("VISA\n");
    }

    else
    {
        printf("INVALID\n");
    }
}

int check_sum(long n)
{
    int r = 0, t = 0;

    do
    {
        t = n % 10;
        r = r + t;
        n = n / 10;
        t = 2 * (n % 10);
        if (t < 10)
        {
            r = r + t;
        }
        else
        {
            r = r + t % 10;
            r = r + t / 10;
        }
        n = n / 10;
    }
    while (n > 0);

    if (r % 10 == 0)
    {
        return 1;
    }
    else
    {
        return 0;
    }
}

int check_length(long n)
{
    int i = 0;
    do
    {
        n = n / 10;
        i++;
    }
    while (n > 1);
    return i;
}

long f_two(long n)
{
    do
    {
        n = n / 10;
    }
    while (n > 100);
    return n;
}

File: done/sentimental-credit/credit.py
from cs50 import get_string


def main():
    n = get_string("Number: ")
    t = n[:2]
    l = len(n)
    if check_sum(n) == 0 or (l != 13 and l != 15 and l != 16):
        print("INVALID")
    elif t == "34" or t == "37":
        print("AMEX")
    elif int(t) > 50 and int(t) < 56:
        print("MASTERCARD")
    elif t[0] == "4":
        print("VISA")
    else:
        print("INVALID")


def check_sum(n):
    c = t = 0
    for i in n[-2::-2]:
        c = int(i) * 2
        if c > 9:
            c = str(c)
            t = t + int(c[0]) + int(c[1])
        else:
            t = t + c
    for i in n[-1::-2]:
        t = t + int(i)
    print(t)
    if t % 10 == 0:
        return 1
    else:
        return 0


main()
